extension/intents/tabs/tabs.js

```diff
@@ -144,4 +144,19 @@ this.intents.tabs = (function() {
       }
     },
   });
+
+  this.intentRunner.registerIntent({
+    name: "tabs.openHomePage",
+    description: "Opens HomePage in a tab",
+    examples: ["open my homepage"],
+    match: `
+    (open | go to |) (my |) homepage (for me|)
+    `,
+    async run(context) {
+      const result = await browser.browserSettings.homepageOverride.get({});
+      result.value
+        .split("|")
+        .map(homepageUrl => browser.tabs.create({ url: homepageUrl }));
```

Comment:
I'm not sure when or why there can be multiple URLs here, so I'm not sure if it's better to open the first URL or open all the URLs. I guess opening them all is an interesting choice at least? Maybe whoever somehow has multiple URLs will then be inspired to give us feedback.

It's not a big concern, but `.forEach()` would be more appropriate here than `.map()`, since it doesn't make use of the return values. But, one issue here is that any errors won't be caught – each `browser.tabs.create()` returns a promise, but nothing watches the promise. You could technically do `await Promise.all(result.value...)`, or just make this a simple loop with awaits.

Reply:
You can put as many URLs as you want in homepage preferences in Firefox that's why I implemented this way. Thanks for the feedback and thanks for sharing your knowledge.